presto-hive/src/main/java/com/facebook/presto/hive/HivePageSource.java

```diff
@@ -752,18 +789,79 @@ public IntArrayList computeEligibleRowIds(Page page)
     @Override
     public boolean pushdownFilterAndProjection(PageSourceOptions options)
     {
+        ImmutableList.Builder<FilterFunction> remainingFilterFunctions = ImmutableList.builder();
+        for (FilterFunction filterFunction : options.getFilterFunctions()) {
+            Set<ColumnMappingKind> inputKinds = getInputKinds(filterFunction);
+            if (!inputKinds.contains(PREFILLED)) {
+                remainingFilterFunctions.add(filterFunction);
+                continue;
+            }
+
+            if (inputKinds.size() > 1) {
+                // TODO Pass necessary pre-filled values to the delegate so it can evaluate these filters
+                throw new UnsupportedOperationException("Filters on a mix of partition and non-partition columns are not supported");
+            }
+
+            if (!evaluateFilterFunction(filterFunction)) {
```

Comment:
Just double checking, this means skip the entire partition, right?  Or is it an error case?

Reply:
Yes, this means skip the entire file.